DIPs/DIP1xxx-JS.md

```diff
@@ -0,0 +1,110 @@
+# The Deprecation Process
+
+| Field | Value |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | xxxx |
+| Review Count:   | 0 |
+| Author:         | Jack Stouffer |
+| Implementation: | N/A |
+| Status:         | Draft |
+
+## Abstract
+
+In order to incrementally improve D or it's standard library, it's often necessary to
+mark features or functions for future removal. This document proposes a standardized
+process for language maintainers to remove public features. This process would be 
+used across DMD, Druntime, and Phobos.
+
+## Rationale
+
+There is general disagreement on the best and/or accepted way to remove public
+features. Each deprecation ends up being handled slightly differently depending on
+who's handling the pull requests. Standardizing the process makes sure that
+deprecations are done very publicly and carefully, so as to minimize breakage and
+to provide clear fixes for user code.
+
+## Description
+
+The following lays out the rules to follow to deprecate a part of D.
+
+A symbol or feature must not be marked for removal on a specific date, but rather on a
+specific release. This allows users to easily know if upgrading will break their
+code or not.
+
+Users must be given at least 10 non-patch releases before the deprecated features
+are removed. More releases can be given if the removed code is commonly used.
+There are two cases where the deprecation period is allowed to be shorter:
+
+1. The code or feature is notably dangerous or unsafe, and users need to remove
+it from their code as soon as possible.
+2. The existance of the current code precludes its own fix or the fix of an equally
+important issue.
+
+Shorting the deprecation preiod should be done with caution to avoid giving D
+an image of instability.
+
+At the time of the pull request for deprecation, all code in Phobos, Druntime,
+and DMD must be updated to remove use of the effected code. Any projects that
```

Comment:
FWIW while I successfully introduced `-de` to DMD, Phobos and Tools, DRuntime doesn't build with `-de` yet (https://github.com/dlang/druntime/pull/1982)
Definitely not a requirement for this DIP, just an FYI.